### ISSUE TITLE
Comments inside an array is not added to AST

### ISSUE THREAD

**Issue by rasmusbe** [glayzzle/php-parser#97]
Hi,

It seems like the comments in these three examples isn't added to the AST.
Examples:
```php
$t = [
	/* translators: sprintf test translation in array */
	sprintf( __( 'sprintf translation in array', 'testdomain' ) ),
];

$t = [
	/* translators: test translation in keyed array */
	'foo' => __( 'translation in keyed array', 'testdomain' ),
];

$message = sprintf(
	/* translators: example inside sprintf */
	__( 'translation inside sprintf', 'testdomain' )
);
```

Parser settings used:
```js
const Engine = require('php-parser');
const parser = new Engine({
  parser: {
    extractDoc: true
  },
  ast: {
    withPositions: true
  }
});
```

Related issue: rasmusbe/wp-pot#29

**Comment by ichiriac** [comment]
Hi Rasmusbe,

I'll try to look at it this week-end but at the time I've started to add doc nodes on AST I've chose to avoid modifying special structures as the node structures becomes more complex to transverse.

If you want to, you can take a look and make a PR, here the array spec :
https://github.com/glayzzle/php-parser/blob/master/docs/AST.md#array

And here where to start to work in order to let pass doc nodes :
https://github.com/glayzzle/php-parser/blob/master/src/parser/array.js#L33

I don't know yet how to provide a clean way to handle docs and avoid breaking changes over ast structure.

**Comment by ichiriac** [comment]
Hi @rasmusbe,

I've new ideas on how to implement documentation nodes / you can see the discussion here : https://github.com/glayzzle/php-parser/issues/126

Are you agree if in the next major release the comments will be stored into a separate array ?

**Comment by ichiriac** [comment]
Hi @rasmusbe,

This issue will be fixed in the next v3.0.0 release (coming soon). Take a look at this thread for more details : https://github.com/glayzzle/php-parser/issues/126#issuecomment-370268380
